applications/fishing-map/src/features/i18n/i18nNumber.tsx

```diff
@@ -1,14 +1,33 @@
 import React, { Fragment } from 'react'
 import { useTranslation } from 'react-i18next'
 import { LocaleOptions } from 'types'
+import i18n from './i18n'
 
 type I18Number = string | number
+type I18NumberOptions =
+  | LocaleOptions
+  | (Intl.NumberFormatOptions & {
+      locale?: LocaleOptions
+      unit?: string
+      unitDisplay?: 'long' | 'short' | 'narrow'
+      notation?: 'standard' | 'scientific' | 'engineering' | 'compact'
+    })
 
-export const formatI18nNumber = (num: I18Number, locale: LocaleOptions) => {
-  const number = num === 'string' ? parseFloat(num) : num
-  return number.toLocaleString(locale, {
-    maximumFractionDigits: number < 10 ? 2 : 0,
-  })
+export const formatI18nNumber = (
+  number: I18Number,
+  options: I18NumberOptions = i18n.language as LocaleOptions
+) => {
+  const locale = typeof options === 'object' ? options.locale || i18n.language : options
+  const parsedNumber = number === 'string' ? parseFloat(number) : (number as number)
+  try {
+    return new Intl.NumberFormat(locale, {
```

Comment:
This is so cool! And support is good in basically every possible browser https://caniuse.com/?search=Intl.NumberFormat

Reply:
Yes! I didn't know it and it was surprising, really nice. Thanks to bring it!